libhb/sync.c

```diff
@@ -487,6 +506,59 @@ static void alignStream( sync_common_t * common, sync_stream_t * stream,
     }
 }
 
+static void alignStreams( sync_common_t * common, int64_t pts )
+{
+    int           ii;
+    hb_buffer_t * buf;
+
+    if (common->job->align_av_start)
+    {
+        int64_t first_pts = AV_NOPTS_VALUE;
+        int     audio_passthru = 0;
+
+        for (ii = 0; ii < common->stream_count; ii++)
+        {
+            sync_stream_t * stream = &common->streams[ii];
+
+            buf = hb_list_item(stream->in_queue, 0);
+            while (buf != NULL && buf->s.start < pts)
+            {
+                hb_list_rem(stream->in_queue, buf);
+                hb_buffer_close(&buf);
+                buf = hb_list_item(stream->in_queue, 0);
+            }
+            if (buf == NULL)
+            {
+                continue;
+            }
+            if (stream->type == SYNC_TYPE_AUDIO &&
+                stream->audio.audio->config.out.codec & HB_ACODEC_PASS_FLAG)
+            {
+                audio_passthru = 1;
+                if (first_pts < buf->s.start)
+                {
+                    first_pts = buf->s.start;
+                }
+            }
+            else if (!audio_passthru)
```

Comment:
Generally the logic is this:
If there is passthru audio, we can't insert silence buffers.  So to align other streams with the passthru audio, we must drop buffers from the other streams (assuming the other streams start before the passthru audio stream).  I do this by looking at all streams and finding the passthru audio first-pts that is *largest*.  Then for all other streams drop any buffers with smaller pts values.  There is one final step that is performed to fill the small gaps that result from dropping frames where possible.  If there are multiple passthru audio tracks, some of the passthru audio tracks may have a small initial gap that we can do nothing about.

If there is no passthru audio track, we can fill all initial gaps by inserting either silence or black frames. I do this by looking at all streams and finding the first-pts that is *smallest*. Then for all other streams, we insert either silence for audio or a black frames for video to fill the gap from the smallest first-pts that was found till the first pts of each stream.

So the conditional you asked about bypasses the "choose the first-pts that is smallest" logic whenever we find passthru audio and forces the "choose passthru audio first-pts that is largest" logic.